app/test_main.py

```diff
@@ -2,2 +1,43 @@
+import pytest
 
-# write your code here
+
+from app import main
+
+def convert_to_human(animal_age: int,
+                     first_year: int,
+                     second_year: int,
+                     each_year: int
+) -> int:
+    if animal_age < first_year:
+        return 0
+    if animal_age < first_year + second_year:
+        return 1
+    return 2 + (animal_age - first_year - second_year) // each_year
+
+
+@pytest.mark.parametrize(
+    "first_year,second_year,each_year_cat,each_year_dog",
+    [
+        (14, 9, 4, 5),
+        (15.1, 9, 4, 5),
+        (15, 8, 4, 5),
+        (15, 9.1, 4, 5),
+        (15, 9, 3, 4),
+        (15, 9, 4.1, 5.1),
+    ],
+    ids=[
+        "14 cat and dog years should convert into 0 human age.",
+        "15 cat and dog years should convert into 1 human age.",
+        "23 cat and dog years should convert into 1 human age.",
+        "24 cat and dog years should convert into 2 human age.",
+        "27/28 cat and dog years should convert into 2 human age.",
+        "28/29 cat and dog years should convert into 3 human age.",
```

Comment:
The test IDs do not accurately describe the expected human age conversions based on the provided animal ages. Ensure that the IDs match the expected results as per the task description.